File: replace_colors.py
import os
import re

directory = "/Users/user/testbasedweb/test-based-nutrition-ui/src"

replacements = {
    # Reds
    r"#7a2a33": "#9f1e13",
    r"#7A2A33": "#9f1e13",
    r"#5c1c24": "#80180f", # Hover state
    r"#5a1a23": "#80180f", # Hover state
    
    # Background / Light Creams
    r"#fcfaf7": "#f9f5f2",
    r"#FCFAF7": "#f9f5f2",
    r"#f8f5f0": "#f9f5f2",
    r"#F8F5F0": "#f9f5f2",
    r"#f2f0e6": "#dbd4c9",
    
    # Dark Creams / Borders
    r"#e9e7dc": "#dbd4c9",
    r"#E9E7DC": "#dbd4c9",
}

for root, _, files in os.walk(directory):
    for file in files:
        if file.endswith(".tsx") or file.endswith(".ts") or file.endswith(".css"):
            filepath = os.path.join(root, file)
            with open(filepath, "r", encoding="utf-8") as f:
                content = f.read()
            
            new_content = content
            for old, new in replacements.items():
                new_content = re.sub(old, new, new_content)
                
            if new_content != content:
                with open(filepath, "w", encoding="utf-8") as f:
                    f.write(new_content)
print("Colors replaced successfully")

File: replace_colors.mjs
import fs from 'fs';
import path from 'path';

const baseDir = path.join(process.cwd(), 'src');

const replacements = [
  // Light Cream
  { from: /#fdfdf9/gi, to: '#f9f5f2' },
  { from: /#f5f5f5/gi, to: '#f9f5f2' },
  { from: /#fdfcfb/gi, to: '#f9f5f2' },
  { from: /#fefdfb/gi, to: '#f9f5f2' },
  { from: /#fdfaf8/gi, to: '#f9f5f2' },
  { from: /#f9f8f4/gi, to: '#f9f5f2' },
  { from: /#F3F2EE/gi, to: '#f9f5f2' },
  { from: /#F7EEE8/gi, to: '#f9f5f2' },
  
  // Dark Cream
  { from: /#d0bfae/gi, to: '#dbd4c9' },
  { from: /#5a3a2d/gi, to: '#dbd4c9' },

  // Red
  { from: /#80180f/gi, to: '#9f1e13' },
  { from: /#8c353f/gi, to: '#9f1e13' },
  { from: /#4a161d/gi, to: '#9f1e13' }
];

function processDir(dir) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    if (stat.isDirectory()) {
      processDir(filePath);
    } else if (filePath.endsWith('.tsx') || filePath.endsWith('.ts') || filePath.endsWith('.css')) {
      let content = fs.readFileSync(filePath, 'utf8');
      let changed = false;
      for (const rule of replacements) {
        if (rule.from.test(content)) {
          content = content.replace(rule.from, rule.to);
          changed = true;
        }
      }
      if (changed) {
        fs.writeFileSync(filePath, content, 'utf8');
        console.log('Updated:', filePath);
      }
    }
  }
}

processDir(baseDir);
